docs/guide/learn/optional-dependencies.md

```diff
@@ -0,0 +1,253 @@
+    ],
+    fields=["analysis"],
+)
+```
+
+## Join Behavior
+
+| Dependency Type             | Join Type  | Behavior                                          |
+| --------------------------- | ---------- | ------------------------------------------------- |
+| Required (`optional=False`) | Inner join | Sample must exist in both features                |
+| Optional (`optional=True`)  | Left join  | Sample preserved even if no match in optional dep |
+
+## When to Use Optional Dependencies
+
+### Enrichment Pattern
+
+Use optional dependencies when you have a **base feature** that should always be present, with **optional enrichment data** that may or may not exist:
+
+```python
+class VideoAnalysis(
+    BaseFeature,
+    spec=FeatureSpec(
+        key="video/analysis",
+        id_columns=["video_id"],
+        deps=[
+            FeatureDep(feature=RawVideo),  # Required - defines sample universe
+            FeatureDep(
+                feature=AudioTranscription, optional=True
+            ),  # Optional enrichment
+            FeatureDep(feature=ManualAnnotations, optional=True),  # Optional enrichment
+        ],
+        fields=["analysis"],
+    ),
+):
+    pass
+```
+
+In this example:
+
+- All samples from `RawVideo` are preserved
+- If `AudioTranscription` exists for a sample, it's included; otherwise, those columns are `NULL`
+- If `ManualAnnotations` exists for a sample, it's included; otherwise, those columns are `NULL`
+
+### Multi-Source Fusion
+
+Use optional dependencies when combining data from **multiple sources** where not all sources have all samples:
+
+```python
+class UnifiedUserProfile(
+    BaseFeature,
+    spec=FeatureSpec(
+        key="user/unified",
+        id_columns=["user_id"],
+        deps=[
+            FeatureDep(feature=CoreUserData),  # Required base
+            FeatureDep(feature=SocialMediaData, optional=True),
+            FeatureDep(feature=PurchaseHistory, optional=True),
+        ],
+        fields=["profile"],
+    ),
+):
+    pass
+```
+
+## Rules
+
+### First Dependency Must Be Required
+
+The **first dependency** in the list defines the "sample universe" - it determines which samples exist in the downstream feature.
+Therefore, the first dependency **cannot be optional**:
+
+```python
+# INVALID - first dependency cannot be optional
+spec = FeatureSpec(
+    key="my/feature",
+    id_columns=["id"],
+    deps=[
+        FeatureDep(feature=OptionalData, optional=True),  # Error!
+        FeatureDep(feature=RequiredData),
+    ],
+    fields=["output"],
+)
+# Raises: ValueError: The first dependency cannot be optional...
+
+# VALID - first dependency is required
+spec = FeatureSpec(
+    key="my/feature",
+    id_columns=["id"],
+    deps=[
+        FeatureDep(feature=RequiredData),  # Required (first)
+        FeatureDep(feature=OptionalData, optional=True),  # Optional (subsequent)
+    ],
+    fields=["output"],
+)
+```
+
+### Join Order Follows Declaration Order
```

Comment:
I don't think this matters given we remove this requirement.

Dependency specification ordering should only affect provenance calculations, not actual data. This is just because provenance calculations are ordered in nature.